src/com/xilinx/rapidwright/design/compare/DesignComparator.java

```diff
@@ -0,0 +1,417 @@
+     * 
+     * @param gold The expected design or reference.
+     * @param test The design being compared with the reference.
+     * @return The total number of differences encountered.
+     */
+    public int compareDesigns(Design gold, Design test) {
+        resetDiffCount();
+        if (!gold.getPartName().equals(test.getPartName())) {
+            addDiff(DesignDiffType.DESIGN_PARTNAME, gold, test, null, "");
+        }
+        
+        if (comparePlacement) {
+            Map<String, SiteInst> goldMap = getSiteInstMap(gold);
+            Map<String, SiteInst> testMap = getSiteInstMap(test);
+
+            for (Entry<String, SiteInst> e : goldMap.entrySet()) {
+                SiteInst testSiteInst = testMap.remove(e.getKey());
+                if (testSiteInst == null) {
+                    addDiff(DesignDiffType.SITEINST_MISSING, e.getValue(), null, gold, "");
+                    continue;
+                }
+                compareSiteInsts(e.getValue(), testSiteInst);
+            }
+
+            for (Entry<String, SiteInst> e : testMap.entrySet()) {
+                addDiff(DesignDiffType.SITEINST_EXTRA, null, e.getValue(), test, "");
+            }
+        }
+
+        Map<String, Net> goldNetMap = getNetMap(gold);
+        Map<String, Net> testNetMap = getNetMap(test);
+
+        for (Entry<String, Net> e : goldNetMap.entrySet()) {
+            Net testNet = testNetMap.remove(e.getKey());
+            if (testNet == null) {
+                addDiff(DesignDiffType.NET_MISSING, e.getValue(), null, gold, "");
+                continue;
+            }
+            compareNets(e.getValue(), testNet);
+        }
+
+        for (Entry<String, Net> e : testNetMap.entrySet()) {
+            addDiff(DesignDiffType.NET_EXTRA, null, e.getValue(), test, "");
+        }
+
+        return getDiffCount();
+    }
+    
+    private Map<String,SiteInst> getSiteInstMap(Design design) {
+        Map<String,SiteInst> map = new HashMap<>();
+        for(SiteInst si : design.getSiteInsts()) {
+            String siteName = si.getSiteName();
+            if (siteName == null) {
+                System.out.println("WARNING: " + design.getName() 
+                    + " has at least one unplaced site instance: " + si.getName());
+            }
+            map.put(si.getSiteName(), si);
+        }
+        return map;
+    }
+
+    private Map<String, Net> getNetMap(Design design) {
+        Map<String, Net> map = new HashMap<>();
+        for (Net net : design.getNets()) {
+            map.put(net.getName(), net);
+        }
+        return map;
+    }
+
+    /**
+     * Gets the total number of differences encountered since the last call of
+     * {@link #compareDesigns(Design, Design)}.
+     * 
+     * @return Total number of design differences found.
+     */
+    public int getDiffCount() {
+        return diffCount;
+    }
+
+    public void resetDiffCount() {
+        diffCount = 0;
+        diffMap = new HashMap<>();
```

Comment:
(will also need `import java.util.EnumMap;`)
```suggestion
        diffMap = new EnumMap<>(DesignDiffType.class);
```

Reply:
Updated